### PR TITLE
:sparkles: Support bootstrap data in secret

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by vincepri** [comment]
The second error seems it's related to converting v1a3 -> v1a2, which isn't supported

File: cloud/scope/machine.go [hunk 1]
```diff
@@ -183,10 +184,29 @@ func (m *MachineScope) SetAnnotation(key, value string) {
...
+// GetBootstrapData returns the bootstrap data from the secret in the Machine's bootstrap.dataSecretName.
+func (m *MachineScope) GetBootstrapData() (string, error) {
+	if m.Machine.Spec.Bootstrap.DataSecretName == nil {
+		return "", errors.New("error retrieving bootstrap data: linked Machine's bootstrap.dataSecretName is nil")
+	}
+
+	secret := &corev1.Secret{}
+	key := types.NamespacedName{Namespace: m.Namespace(), Name: *m.Machine.Spec.Bootstrap.DataSecretName}
+	if err := m.client.Get(context.TODO(), key, secret); err != nil {
+		return "", errors.Wrapf(err, "failed to retrieve bootstrap data secret for AWSMachine %s/%s", m.Namespace(), m.Name())
```
Comment (ncdc) [inline cloud/scope/machine.go]: AWSMachine

Comment (vincepri) [reply cloud/scope/machine.go]: fixed, thanks!

**Pull request merged**